app/controllers/mixins/actions/vm_actions/resize.rb

```diff
@@ -3,16 +3,26 @@ module Actions
     module VmActions
       module Resize
         def resizevms
-          assert_privileges("instance_resize")
-          recs = find_checked_items
+          assert_privileges(params[:pressed])
+          # if coming in to edit from miq_request list view
+          recs = checked_or_params
+          if !session[:checked_items].nil? && (@lastaction == "set_checked_items" || params[:pressed] == "miq_request_edit")
+            request_id = params[:id]
+            @record = VmCloudReconfigureRequest.find(request_id).vms.first
+          end
+
           recs = [params[:id].to_i] if recs.blank?
-          @record = find_record_with_rbac(VmOrTemplate, recs.first) # Set the VM object
+          @record ||= find_record_with_rbac(VmOrTemplate, recs.first) # Set the VM object
```

Comment:
This breaks any code where @record was previously set, are you sure none such exists?

Reply:
@record may have been set a few lines above, in which case we want to keep it. This is to handle the case where it was not set. There is a possible edge case if @record is not set above within this method but is already set before -- we could handle that by setting @record to nil before the if block that optionally sets it.